source/common/router/config_impl.cc

```diff
@@ -1012,7 +1017,8 @@ RouteConstSharedPtr VirtualHostImpl::getRouteFromEntries(const Http::HeaderMap&
 
 const VirtualHostImpl* RouteMatcher::findVirtualHost(const Http::HeaderMap& headers) const {
   // Fast path the case where we only have a default virtual host.
-  if (virtual_hosts_.empty() && wildcard_virtual_host_suffixes_.empty() && default_virtual_host_) {
+  if (virtual_hosts_.empty() && wildcard_virtual_host_suffixes_.empty() &&
+      wildcard_virtual_host_prefixes_.empty() && default_virtual_host_) {
```

Comment:
I wonder if we even need the `default_virtual_host_` clause here. We end up returning `default_virtual_host_.get()` at the end anyway.

Reply:
You're right, clause removed.